### PR TITLE
Add in CodeQL Analysis and check .net is in LTS

### PR DESCRIPTION
**Pull request by jdsmithit**
Adds a GitHub action that will scan for vulnerabilities also checks.net Framework is in LTS.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
```
Comment (avnes) [inline .github/workflows/codeql-analysis.yml]: For easier reuse, it is perhaps better to use:

```
branches:
  - master
  - main
```

**Review by avnes — DISMISSED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (jdsmithit) [reply .github/workflows/codeql-analysis.yml]: Great Suggestion!

Comment (jdsmithit) [reply .github/workflows/codeql-analysis.yml]: done also added the sweeper check on builds and pushes

Comment (jdsmithit) [reply .github/workflows/codeql-analysis.yml]: @avnes Done

**Review by avnes — APPROVED** [review]

**Pull request merged**